packages/scene-composer/src/components/three-fiber/DataOverlayComponent/__tests__/DataOverlayContainerSnap.spec.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import { IDataOverlayComponentInternal, useStore } from '../../../../store';
+import { KnownComponentType } from '../../../../interfaces';
+import { Component } from '../../../../models/SceneModels';
+import { DataOverlayContainer } from '../DataOverlayContainer';
+
+jest.mock('../../../../hooks/useCallbackWhenNotPanning', () => (cb) => [
+  jest.fn(),
+  function Hack(e) {
+    cb(e);
+  },
+]);
+
+jest.mock('../DataOverlayRows', () => ({
+  DataOverlayRows: (...props: any[]) => <div data-testid='rows'>{JSON.stringify(props)}</div>,
```

Comment:
that's to prevent us from updating the snapshot of all the parent component when the child component is updated. this component is quite static, mostly just rendering different content based on props, so each component making sure it renders the child with expected props will be good enough